tools/run_tests/xds_k8s_test_driver/framework/infrastructure/gcp/compute.py

```diff
@@ -80,6 +80,34 @@ def create_health_check(self,
     def delete_health_check(self, name):
         self._delete_resource(self.api.healthChecks(), 'healthCheck', name)
 
+    def create_firewall_rule(self, name: str, network_url: str,
+                             source_ranges: List[str],
+                             ports: List[str]) -> Optional[GcpResource]:
+        try:
+            return self._insert_resource(
+                self.api.firewalls(), {
+                    "allowed": [{
+                        "IPProtocol": "tcp",
+                        "ports": ports
+                    }],
+                    "direction": "INGRESS",
+                    "name": name,
+                    "network": network_url,
+                    "priority": 1000,
+                    "sourceRanges": source_ranges,
+                    "targetTags": ["allow-health-checks"]
+                })
+        except googleapiclient.errors.HttpError as http_error:
+            # TODO(lidiz) use status_code() when we upgrade googleapiclient
+            if http_error.resp.status == 409:
+                logger.debug('Firewall rule %s already existed', name)
```

Comment:
```suggestion
                logger.info('Firewall rule %s already exists', name)
```
Since there's several possible outcomes (created vs exists), let's make it easier to differentiate between them.